src/python/pants/python/python_setup.py

```diff
@@ -215,6 +223,102 @@ def get_pex_python_paths():
         else:
             return []
 
+    @staticmethod
+    def get_asdf_paths(env: Environment, *, asdf_local: bool = False) -> List[str]:
+        """Returns a list of paths to Python interpreters managed by ASDF.
+
+        :param env: The environment to use to look up ASDF.
+        :param bool asdf_local: If True, only use the interpreter specified by
+                                '.tool-versions' file under `build_root`.
+        """
+        asdf_dir = get_asdf_dir(env)
+        if not asdf_dir:
+            return []
+
+        asdf_dir = Path(asdf_dir)
+
+        # Ignore ASDF if the python plugin isn't installed
```

Comment:
We prefer for comments to end in a period: https://www.pantsbuild.org/docs/style-guide#style

There are a couple places it'd be good to add that, please.